app/controllers/variable_drafts_controller.rb

```diff
@@ -16,6 +16,25 @@ def edit
     set_measurement_names if @current_form == 'measurement_identifiers'
   end
 
+  def new
+    unless params[:associated_collection_id].blank?
+      @associated_collection_id = params[:associated_collection_id].strip
+      current_collection_response = cmr_client.get_collections_by_post({ concept_id: @associated_collection_id }, token)
+
+      if current_collection_response.success? && current_collection_response.body['hits'] > 0 && current_provider?(current_collection_response.body.dig('items',0,'meta','provider-id'))
+        super
+      elsif !current_collection_response.success?
+        redirect_to manage_variables_path, flash: { error: current_collection_response.body['errors'].first }
+      elsif current_collection_response.body['hits'] == 0
+        redirect_to manage_variables_path, flash: { error: "No matches were found for #{@associated_collection_id}" }
+      elsif !current_provider?(current_collection_response.body.dig('items',0,'meta','provider-id'))
+        redirect_to manage_variables_path, flash: { error: "Variables can only be associated to collections within the same provider. To create a variable for #{@associated_collection_id} you must change your provider context." }
+      end
+    else
+      super
+    end
+  end
+
```

Comment:
@charleshuang80 @ryanmiller-1 I have given this a bit of thought and there are 2 things that I'm unsure about: 

1. Regarding the **first two** `elsif` statements; the first one was originally meant to flash a message when the call to cmr was unsuccessful (network error, invalid permission, whatever). However, since the `current_collection_response.success?` is `false` when a user enters jibberish, for instance, this infringes on the purpose of the next `elsif` statement which is to flash a message when the supplied concept-id is not found in CMR. To rectify this, all I've done is surface the cmr error response. I was hoping to get your take on the overlap between these two conditionals and if it even matters. 
2. I am displeased with the readability of this whole controller method because of the nested conditionals but I can't think of a simpler way to do this. I was hoping you guys might have advice for this sort of thing as well.

Reply:
Also, FWIW, I think that the number of times a user enters an invalid concept_id will be very small.

Reply:
- I like that, definitely going to implement the guard clause (I'll make sure to check it as well)
- Okay that makes sense, I will leave it
- Yeah I suppose so.

Reply:
For 1: I think surfacing a CMR error is sufficient, we would like their error messages to be as user friendly as possible and should avoid having to interpret them for the user. For the case of 0 hits, did you run that language by the PO? I feel like we might want a tiny bit more information in that error message, but if it was approved by PO then that's fine.

For 2: I agree a guard close would help. I would actually add a second guard clause, checking for the current provider. I don't think we even need to make the call to CMR to verify the collection if it's the wrong provider.

Reply:
1 - Yeah, PO requested it; originally had something different.

2 - That's a great point, will do!

Reply:
so the way that a collection concept id is constructed is `C` `<number>` `-` `<provider_id>`. I think you can split it on the `-` and assume the last part is the provider id. But if you think it is safer or would prefer to use the provider id from the response you don't have to add the guard clause the way I suggested.

Reply:
Honestly, after thinking about changing this, I am on the fence. Your suggestion would work fine, I really can't find any flaws - it would remove one of the conditionals and be easy to implement. However, I am on the fence because we don't really do this to get the `provider-id` elsewhere. What do you think I should do?

Reply:
@charleshuang80

Reply:
Leaving it as is is definitely a bit 'safer'. But I lean towards adding the additional guard clause - the concept id construction doesn't change and I believe we use it for groups (esp verifying system groups) and validating subscriptions.